Source/monster.cpp

```diff
@@ -1949,50 +1949,51 @@ void GroupUnity(int i)
 	assert(i >= 0 && i < MAXMONSTERS);
 	auto &monster = Monsters[i];
 
+	if (monster.leaderflag == MonsterRelation::Individual)
+		return;
+
+	assert(monster.leader >= 0);

```

Comment:
arh, this explains it, I thought this function was called both for the leader and the minions. Thanks for spending the time to correct this.

Reply:
Initially I thought the same 😄 
Then my debugger told me otherwise. First I thought wtf and added debug output for `leader`/`minion` flag and was a little bit surprised. 😁